src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/QinQFlowSpec.groovy

```diff
@@ -885,4 +887,80 @@ class QinQFlowSpec extends HealthCheckSpecification {
         cleanup:
         flow && flowHelperV2.deleteFlow(flow.flowId)
     }
+
+    @Tidy
+    def "System doesn't reroute flow to more preferable path while updating innerVlanId"() {
+        given: "Two active switches connected to traffgens with two possible paths at least"
+        def allTraffgenSwitchIds = topology.activeTraffGens*.switchConnected*.dpId ?:
+                assumeTrue("Should be at least two active traffgens connected to switches", false)
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].every { sw ->
+                sw.dpId in allTraffgenSwitchIds && northbound.getSwitchProperties(sw.dpId).multiTable
+            } && it.paths.size() > 2
+        } ?: assumeTrue("Not able to find enough switches with traffgens and in multi-table mode", false)
+
+        and: "A flow"
+        def flow = flowHelperV2.randomFlow(switchPair)
+        flow.source.innerVlanId = flow.source.vlanId
+        flow.destination.innerVlanId = flow.destination.vlanId
+        flowHelperV2.addFlow(flow)
+
+        when: "Make the current path less preferable than alternatives"
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flow.flowId))
+        def alternativePaths = switchPair.paths.findAll { it != currentPath }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentPath) }
+
+        and: "Update the flow: port number and vlanId on the src/dst endpoints"
+        def updatedFlow = flow.jacksonCopy().tap {
+            it.source.innerVlanId = flow.destination.vlanId
+            it.destination.innerVlanId = flow.source.vlanId
+        }
+        flowHelperV2.updateFlow(flow.flowId, updatedFlow)
+
+        then: "Flow is really updated"
+        with(northboundV2.getFlow(flow.flowId)) {
+            it.source.portNumber == updatedFlow.source.portNumber
```

Comment:
the same issue, btw, no inner vlan validation here

Reply:
thx, fixed